src/Kestrel.Transport.Abstractions/Internal/TransportConnection.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Net;
+using System.Text;
+using Microsoft.AspNetCore.Protocols.Features;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal
+{
+    public abstract partial class TransportConnection
+    {
+        public TransportConnection()
+        {
+            _currentIConnectionIdFeature = this;
+            _currentIConnectionTransportFeature = this;
+            _currentIHttpConnectionFeature = this;
+        }
+
+        public IPAddress RemoteAddress { get; set; }
```

Comment:
Why do this pattern instead of just using the same property names as `IHttpConnectionFeature`, etc. ?